src/components/shared/MarkerTooltipContents.js

```diff
@@ -34,6 +47,50 @@ function getMarkerDetails(data: MarkerPayload): React$Element<*> | null {
           </div>
         );
       }
+      case 'GCMinor': {
+        if (data.nursery && data.nursery.reason) {
```

Comment:
Flow types are just erased, there is no difference in the JS code before/after the erasure. So it _is_ possible to assign null to a variable that, according to flow types, can't be null.
That's why we push quite hard to have a complete flow coverage in the project, as incomplete coverage can bring issues like this not being caught by flow :/

Regarding your planned change to the markers, you'll also need to upgrade the format and add conversion steps in both https://github.com/devtools-html/perf.html/blob/master/src/profile-logic/gecko-profile-versioning.js (so that we can still read profiles from older Firefox) and https://github.com/devtools-html/perf.html/blob/master/src/profile-logic/processed-profile-versioning.js (so that we can still read older uploaded profiles with such markers).